tests/v2/integration/resource_quota_test.go

```diff
@@ -0,0 +1,284 @@
+
+	quotas, err = resourcequotas.ListResourceQuotas(client, "local", "ns2", metav1.ListOptions{})
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), quotas)
+	require.Lenf(s.T(), quotas.Items, 1, "Expected 1 quota in ns2, but got %d", len(quotas.Items))
+
+	limit = &v3.ResourceQuotaLimit{
+		LimitsCPU: "0",
+	}
+
+	resourceList = quotas.Items[0].Spec.Hard
+	rl, err = v3quotas.ConvertProjectResourceLimitToResourceList(limit)
+	require.Equal(s.T(), rl, resourceList)
+	require.Equal(s.T(), limit, limit)
+	require.NoError(s.T(), err)
+}
+
+func (s *ResourceQuotaSuite) TestRemoveQuotaFromProjectWithNamespacePropagation() {
+	subSession := s.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := s.client.WithSession(subSession)
+	require.NoError(s.T(), err)
+
+	projectLimit := &management.ResourceQuotaLimit{
+		LimitsCPU:  "500m",
+		ConfigMaps: "10",
+	}
+	namespaceDefaultLimit := &management.ResourceQuotaLimit{
+		LimitsCPU:  "200m",
+		ConfigMaps: "5",
+	}
+
+	projectConfig := &management.Project{
+		ClusterID: "local",
+		Name:      "TestProject",
+		ResourceQuota: &management.ProjectResourceQuota{
+			Limit: projectLimit,
+		},
+		NamespaceDefaultResourceQuota: &management.NamespaceResourceQuota{
+			Limit: namespaceDefaultLimit,
+		},
+	}
+	testProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(s.T(), err)
+
+	namespace, err := namespaces.CreateNamespace(client, "ns1", "", map[string]string{}, map[string]string{}, testProject)
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), namespace)
+
+	testProject.ResourceQuota.Limit.LimitsCPU = ""
+	testProject.NamespaceDefaultResourceQuota.Limit.LimitsCPU = ""
+
+	_, err = client.Management.Project.Replace(testProject)
+	require.NoError(s.T(), err)
+
+	// Allow the controller to update the resource quotas after the project has been updated.
+	time.Sleep(2 * time.Second)
+
+	quotas, err := resourcequotas.ListResourceQuotas(client, "local", "ns1", metav1.ListOptions{})
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), quotas)
+	require.Lenf(s.T(), quotas.Items, 1, "Expected 1 quota in the namespace, but got %d", len(quotas.Items))
+
+	limit := &v3.ResourceQuotaLimit{
+		ConfigMaps: "5",
+	}
+
+	resourceList := quotas.Items[0].Spec.Hard
+	rl, err := v3quotas.ConvertProjectResourceLimitToResourceList(limit)
+	require.Equal(s.T(), rl, resourceList)
+	require.NoError(s.T(), err)
+
+	// Now remove the last resource limit from the project.
+	testProject.ResourceQuota.Limit.ConfigMaps = ""
+	testProject.NamespaceDefaultResourceQuota.Limit.ConfigMaps = ""
+
+	_, err = client.Management.Project.Replace(testProject)
+	require.NoError(s.T(), err)
+
+	time.Sleep(2 * time.Second)
+	quotas, err = resourcequotas.ListResourceQuotas(client, "local", "ns1", metav1.ListOptions{})
+	require.NoError(s.T(), err)
+	require.NotNil(s.T(), quotas)
+	require.Lenf(s.T(), quotas.Items, 1, "Expected 1 quota in the namespace, but got %d", len(quotas.Items))
```

Comment:
This isn't what I expected. Why aren't we triggering https://github.com/rancher/rancher/blob/acb99e9d0aefb6fcca70963beba3423b8898a49d/pkg/controllers/managementuser/resourcequota/resource_quota_sync.go#L207-L210 ?

Reply:
For some reason, there is still one nil resource quota, investigating...

Reply:
I noticed that one can create a regular K8s resource quota without specifying any spec and thus with no limits or requests.
The API server allows it and creates an empty quota with a given name.
What happens here is similar. Perhaps we should delete a quota object if its spec is empty?

Reply:
> Perhaps we should delete a quota object if its spec is empty?

That's what I was getting at. Why isn't this happening? If it's empty, what could a user possibly need from it?

Reply:
This is old behavior, by the way. I will see how to delete the quota if its resource list is empty.

Reply:
OK, an empty quota remains because the annotation is always updated on the namespace. In cases like these, it looks like this: `field.cattle.io/resourceQuota: '{"limit":{}}'`.
Should we just delete the annotation together with the quota object? Is the namespace in a valid state then?

Reply:
Made an update. If we delete the quota but keep the annotation, then there they are out of sync, so I delete both together. Tested this by adding and removing different resources from the project and seeing how the annotation and quota get updated.